docs: :memo: installation guide on Sprout with pip (#765)
seedcase-project/seedcase-sprout: docs/guide/installation.qmd

---
title: "Installing Sprout"
order: 0
---

We designed Sprout to be installed for use either on your own computer
(locally) or on a server (remote). This guide will walk you through
installing Sprout. Regardless of how you install Sprout, you will need

> to use the Terminal. If you unfamiliar with the terminal, start with our

Reviewer: ```suggestion
to use the Terminal. If you are unfamiliar with the terminal, start with our
```

[guide for beginners]() before continuing.

<!-- TODO: this callout might need to be its own include, since other docs will use it too -->

::: callout-important
Sprout as a Python package and as a CLI assumes you have full control
over the folders and files of the system, or at least your user's home
directory. This includes being given space on a server that mostly has
access through a Terminal, where you have control over the directories
you can write to.
:::

## Use in a project in a virtual environment

Sprout requires [Python](https://www.python.org/downloads/) version

> \>=3.6 be installed on your system, which includes pip. Run this command

Reviewer 2: There isn't a need to separately install pip, it comes with Python.

in the Terminal to check your version of Python:

``` bash
python3 --version
```

If the version isn't 3.6 or greater, download and install the latest
version of Python by going to their [download
page](https://www.python.org/downloads/).

> ### Creating a virtual environment

Reviewer 2: We needed the section on creating a virtual environment, since it isn't recommended to install via pip into the global environment.

In general, it isn't recommended to use pip to install into your global
system (accessible everywhere on your computer) because it can

> negatively effect other Python packages. Rather, you install a package

Author: affect

> negatively effect other Python packages. Rather, you install a package

Reviewer 2: ```suggestion
negatively affect other Python packages. Rather, you install a package
```

into a virtual environment so that it doesn't affect your global system.

To make this virtual environment, we need to create it with venv in a
folder location that makes sense for your purposes. A good location
might be in your documents folder or desktop. For now, we will create it
on the desktop:

``` bash
cd Desktop
mkdir use-sprout
python3 -m venv .venv
source .venv/bin/activate
```

<!-- TODO: Write instructions using Poetry as well -->

You're now ready to install Sprout!

### Install Sprout

We can install Sprout directly from the [GitHub
repository](https://github.com/seedcase-project/seedcase-sprout) by
running the following command:

``` bash
pip install git+https://github.com/seedcase-project/seedcase-sprout
```

To check that Sprout has been installed correctly, run the following
command in your terminal:

``` bash

> pip show seedcase_sprout

Reviewer 2: This is a better way to test if it installed, plus it shows the version and other details.

```

If it installed successful, it will show details about Sprout.

Get started with your first data package by following the guide on
[Creating and managing data packages](/docs/guide/packages.qmd).

> ## Use anywhere on the computer or in the Terminal

Reviewer 2: This section will include instructions using pipx.

If you're not comfortable using the terminal, fear not! Below you'll find a thorough
step-by-step guide to help you along.

<!-- TODO: Add instructions for people not comfortable with the terminal -->
:::
### 1. **Python** and **pip**

Sprout requires Python and pip to be installed on your system:

::: {.callout-note title="Software requirements"}
-   [Python](https://www.python.org/downloads/) 3.6–3.12
-   [pip](https://pip.pypa.io/en/stable/installation/) version 19.0 or
    higher
    <!--TODO: Do we have a specific pip version we require? This version
    was taken from TensorFlow's installation guide. They do vary their
    version requirements based on OS, not sure if we should too?-->
:::

Check your Python and pip version by running the following commands in
your terminal:

``` bash
python3 --version
python3 -m pip --version
```

If you don't get a Python version between 3.6 and 3.12 (or at all), go
to the [Python website](https://www.python.org/downloads/) to download
and install Python.

If you don't have pip installed, you can install it by following the
instructions on the [pip
website](https://pip.pypa.io/en/stable/installation/).

> ### 2. **Install Sprout**

Reviewer 2: ```suggestion
### Install Sprout
```

To install Sprout, make sure that pip is up to date by running the
following command in your terminal:

``` bash
pip install --upgrade pip
```

Then, install Sprout directly from the [GitHub
repository](https://github.com/seedcase-project/seedcase-sprout) by
running the following command:

``` bash
pip install seedcase_sprout@git+https://github.com/seedcase-project/seedcase-sprout
```

### 3. **Check the installation**

To check that Sprout has been installed correctly, run the following
command in your terminal:

``` bash

> python3 -c "import sprout.core as sp; print(sp.PackageProperties())"

Reviewer 2: ```suggestion
pip show seedcase_sprout
```

This shows information about the package, if it is installed.

```

If a `PackageProperties` object is returned, Sprout has been installed
successfully. 

Get started with your first data package by following the guide on 

> [Creating and managing data packages](https://sprout.seedcase-project.org/docs/guide/packages).

Reviewer 2: ```suggestion
[Creating and managing data packages](/docs/guide/packages.qmd).
```

---
title: "Installing Seedcase Sprout using pip"
---

> {{< include /docs/includes/_wip.qmd >}}

Reviewer 3: ```suggestion
```
This file has been deleted and the message is included directly in `_quarto.yml` instead 👍

> In order for Seedcase Sprout to work it will need to be installed either
> locally or on a server. This guide will show you how to install Sprout
> using pip. The requirements are that you have Python 3.6 or higher
> installed on your system, and that you are comfortable with using the
> command line.

Reviewer 3: This might be a bit controversial, but I'm not sure that we need this? :)
```suggestion
```

Author: Depends on what you think your audience is, if it is me a year back, then I would have appreciated the info.

Reviewer 3: I see what you mean. The reasons why I removed this was that I thought it was a bit repetitive (the requirements are listed below) and I didn't want to scare people away who do not feel comfortable using the terminal :) 

I added the "are you not comfortable with the command line?" block as a place to put a more thorough step-by-step guide for how to install Sprout using the terminal, but this might be overkill? I thought about the things you said the other day about when you're new to the terminal, you might not think about which directory you're in before installing, for example :)

If we get to a stage with a non-terminal installation, I think it should have its own section 🌱

Author: Ah - that was how you wanted to use the block, no, that makes perfect sense. The only caveat would be that the guide for terminal novices would be rather long, do we want it on a separate page?

<!-- TODO: this callout might need to be its own include, since other docs will use it too -->

::: callout-important
Sprout assumes you have full control over the folders and files of the
system, or at least your user's home directory. This includes being
given space on a server that mostly has access through a Terminal, where
you have control over the directories you can write to.
:::

> ## Installing Sprout using pip

Reviewer 3: ```suggestion
## Install Sprout using pip
```

> 

Reviewer 3: We could add a collapsed callout block here for people who aren't comfortable with the terminal :)

```suggestion

::: {.callout-tip title="Not too comfortable using the terminal? Look here!" collapse="true"}
If you're not comfortable using the terminal, fear not! Below you'll find a thorough
step-by-step guide to help you along.

<!-- TODO: Add instructions for people not comfortable with the terminal -->
:::
```

### Making sure your system is ready

There are some things you will need to ensure are installed on your
system before you can install Sprout. You will need python 3.6 or

> higher, pip, and pipenv.

Reviewer 3: I don't think we need `pipenv`? :)

Author: @lwjohnst86 I think you were the one to mention pipenv, but now that I think about it, I believe Signe is right..

Once you have python installed (see the [Python

> website](https://www.python.org/downloads/) if you don't have it on your
> machine), you can check if you have pip installed by running the
> following command in your terminal:
> 
> ``` bash
> pip --version
> ```
> 
> The reply to this should be a version number. If you do not have pip
> installed, you can install it by following the instructions on the [pip
> website](https://pip.pypa.io/en/stable/installation/). If pip is
> outdated it will tell you, and provide guidance on how to update it.
> 
> You will also need to check that pipenv is installed. You can do this by
> running the following command in your terminal:
> 
> ``` bash
> pipenv --version
> ```
> 
> The reply to this should be a version number. If you do not have pipenv
> installed, you can install it by running the following command in your
> terminal:
> 
> ``` bash
> pip install pipenv
> ```
> 
> This installation will take a little while, but once it is done you will
> be able to install Sprout.
> 
> ### Installing Sprout
> 
> ::: panel-tabset
> ### Local installation
> 
> Start by deciding where you want to store your data packages. By
> default, Sprout will be installed in the directory that your terminal is
> currently in. If you want to install Sprout in a different directory,
> you can navigate to that directory in your terminal before running the
> installation command (if you don't know how to navigate directories in
> your terminal, you can find a guide
> [here](https://www.digitalcitizen.life/command-prompt-how-use-basic-commands)).
> 
> Once you have decided where you want to install Sprout, you can run the
> following command in your terminal:
> 
> ``` bash
> pip install seedcase_sprout@git+https://github.com/seedcase-project/seedcase-sprout
> ```
> 
> This will install Sprout on your system. Once the installation is
> complete, you will be able to run Sprout from the command line.
> 
> ### Server installation
> 
> ::: callout-warning
> In development.
> :::
> :::

Reviewer 3: Taking inspiration from Tensorflow and poetry's installation guides, I have made an alternative version of how this could look: 

This if ofc just a suggestion ☺️ 

````suggestion
### 1. **Python** and **pip**

Sprout requires Python and pip to be installed on your system:

::: {.callout-note title="Software requirements"}
-   [Python](https://www.python.org/downloads/) 3.6–3.12
-   [pip](https://pip.pypa.io/en/stable/installation/) version 19.0 or
    higher
    <!--TODO: Do we have a specific pip version we require? This version
    was taken from TensorFlow's installation guide. They do vary their
    version requirements based on OS, not sure if we should too?-->
:::

Check your Python and pip version by running the following commands in
your terminal:

``` bash
python3 --version
python3 -m pip --version
```

If you don't get a Python version between 3.6 and 3.12 (or at all), go
to the [Python website](https://www.python.org/downloads/) to download
and install Python.

If you don't have pip installed, you can install it by following the
instructions on the [pip
website](https://pip.pypa.io/en/stable/installation/).

### 2. **Install Sprout**

To install Sprout, make sure that pip is up to date by running the
following command in your terminal:

``` bash
pip install --upgrade pip
```

Then, install Sprout directly from the [GitHub
repository](https://github.com/seedcase-project/seedcase-sprout) by
running the following command:

``` bash
pip install seedcase_sprout@git+https://github.com/seedcase-project/seedcase-sprout
```

### 3. **Check the installation**

To check that Sprout has been installed correctly, run the following
command in your terminal:

``` bash
python3 -c "import sprout.core as sp; print(sp.PackageProperties())"
```

If a `PackageProperties` object is returned, Sprout has been installed
successfully. 

Get started with your first data package by following the guide on 
[Creating and managing data packages](https://sprout.seedcase-project.org/docs/guide/packages).
```
